luna/featurevis/transformations.py

```diff
@@ -152,10 +293,98 @@ def color_augmentation(image, color_aug):
         color_aug (bool): True if image needs to augment.
 
     Returns:
-        img (list): Augmented image.
+        list: Augmented image.
     """
     if color_aug:
         image = tf.image.random_saturation(image, 0, 0.5)
         image = tf.image.random_contrast(image, 0, 0.8)
         image = tf.image.random_brightness(image, 0.01)
     return image
+
+
+def standard_transforms(img):
+    """Standard transformations if no transformations were chosen by user (Suggested by Lucid)
+
+    Args:
+        img (list): Image
+
+    Returns:
+        list: transformed imaged with standard modes
+    """
+    img = pad(img, 12, pad_mode="constant")
+    img = jitter(img, 8)
+    img = bilinear_rescale(img, [1 + (i - 5) / 50.0 for i in range(11)], seed=None)
+    img = rotation(img, list(range(-10, 11)) + 5 * [0])
+    img = jitter(img, 4)
+    return img
+
+
+def perform_trans(img, trans_param):
+    """Perfrom a sequence of transformations
+
+    Args:
+        img (list): image
+        trans_param (class): an instance of the TransformationParameters class
+
+    Returns:
+        list: transformed image
+    """
+
+    img = pad(img, trans_param.pad_size, trans_param.pad_mode)
+    img = jitter(img, trans_param.add_jitter)
+    img = bilinear_rescale(img, trans_param.rescale_val)
+    img = rotation(img, trans_param.angles)
+
+    return img
+
+
+def perform_aux_trans(img, aux_trans_param):
+    """Perfrom a sequence of auxiliary transformations
+
+    Args:
+        img (list): image
+        aux_trans_param (class): an instance of the AuxiliaryTransformationParameters class
+
+    Returns:
+        list: transformed image
+    """
+    img = crop_or_pad(img, aux_trans_param.pad_crop)
+    img = noise(img, aux_trans_param.add_noise)
+    img = rescale(img, aux_trans_param.scale)
+    img = blur(img, aux_trans_param.add_blur)
+    img = flip(img, aux_trans_param.add_flip)
+    img = vert_rotation(img, aux_trans_param.add_rotation)
+    img = color_augmentation(img, aux_trans_param.color_aug)
+
+    return img
+
+
+def perform_custom_trans(img, custom_trans):
+    """Perfrom a sequence of customized transformations
+
+    Args:
+        img (list): image
+        custom_trans (dict): A dictionary of custom transformations where the key is the name of
+                             transformation (same as function name) and the value is the parameter
+                             or list of parameters.
+
+    Raises:
+        ValueError: if the called transformation does not match the function name.
+        ValueError: if the given custom transformation is not a dictionary.
+
+    Returns:
+        list: transformed image.
+    """
+    if isinstance(custom_trans, dict):
+        for key, value in custom_trans.items():
+            if key not in globals():
```

Comment:
We should not do this. Can't we pass the functions themselves in a list? Then we would not have to resolve the dict. See [this on passing functions](https://www.geeksforgeeks.org/passing-function-as-an-argument-in-python/).

Reply:
For this one, what we could do is require a transformation function. Users would write something like this:

```
def my_trans(img):
     img = crop_or_pad(img, aux_trans_param.pad_crop)
     img = noise(img, aux_trans_param.add_noise)
     img = rescale(img, aux_trans_param.scale)
     img = blur(img, aux_trans_param.add_blur)
     img = flip(img, aux_trans_param.add_flip)
     img = vert_rotation(img, aux_trans_param.add_rotation)
     img = color_augmentation(img, aux_trans_param.color_aug)

     return img
```

and pass this function instead of the dict with keys

Reply:
Good solution! It makes the transformation classes (as well as my weekend effort xD) redundant. The Documentation should be updated.